### PR TITLE
Fix Spring App Configuration User Agent

### PR DESCRIPTION
**Pull request by mrm9084**
Updates the pom.xml to include the manifest information to correctly generate the User Agent.

### CODE REVIEW

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/pom.xml [hunk 1]
```diff
@@ -172,6 +172,18 @@
           </rules>
         </configuration>
       </plugin>
+      <plugin>
```
Comment (mssfang) [inline sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/pom.xml:175]: @srinathnarayanan 
Does this change need to go to parent pom?

File: sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/pom.xml [hunk 2]
```diff
@@ -172,6 +172,18 @@
...
+        <groupId>org.apache.maven.plugins</groupId>
```
Comment (mssfang) [inline sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/pom.xml:176]: @srnagar 

Does this change missing something. The PR build failed at verifying version.

Comment (srnagar) [reply sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/pom.xml:176]: @mssfang This is missing a `<version></version>` tag.

Comment (mrm9084) [reply sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/pom.xml:176]: @srnagar As the version was listed as managed.

Comment (srnagar) [reply sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/pom.xml:176]: Version should still be explicitly specified to not use the managed version and instead use the version we use for all client libraries.

Comment (alzimmermsft) [reply sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/pom.xml:176]: `<version>3.1.2</version> <!-- {x-version-update;org.apache.maven.plugins:maven-jar-plugin;external_dependency} -->`

**Pull request merged**